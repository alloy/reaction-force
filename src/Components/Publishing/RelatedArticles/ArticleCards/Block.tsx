import { unica } from "Assets/Fonts"
import { media } from "Components/Helpers"
import {
  Vertical,
  VerticalOrSeriesTitle,
} from "Components/Publishing/Sections/VerticalOrSeriesTitle"
import {
  SeriesAbout,
  SeriesAboutContainer,
} from "Components/Publishing/Series/SeriesAbout"
import { ArticleData } from "Components/Publishing/Typings"
import { MaxRow } from "Components/Publishing/Video/Shared"
import React from "react"
import { Col } from "react-styled-flexboxgrid"
import styled from "styled-components"
import { ArticleCards } from "./ArticleCards"

interface Props {
  article?: ArticleData
  color?: string
  relatedArticles?: any
}

export const ArticleCardsBlock: React.SFC<Props> = props => {
  const { article, color, relatedArticles } = props
  const { seriesArticle } = article

  return (
    <ArticleCardsContainer color={color}>
      {(relatedArticles || article.relatedArticles) && (
        <MaxRow>
          <Col>
            <VerticalOrSeriesTitle
              article={article}
              color={color}
              prefix="More in "
            />
          </Col>
          <ArticleCards
            relatedArticles={relatedArticles || article.relatedArticles}
            series={seriesArticle}
            color={color}
          />
        </MaxRow>
      )}
      {seriesArticle && (
        <MaxRow>
          <SeriesAbout article={seriesArticle} color={color} />
        </MaxRow>
      )}
    </ArticleCardsContainer>
  )
}

ArticleCardsBlock.defaultProps = {
  color: "black",
}

export const ArticleCardsContainer = styled.div`
  color: ${props => props.color};

  ${MaxRow} {
    margin: auto;
  }

  ${Vertical} {
    ${unica("s32")};
    width: 100%;
    margin-bottom: 40px;

    a {
      border-bottom: 2px solid;
      ${media.sm`
        display: block;
      `};
    }
  }

  ${SeriesAboutContainer} {
    margin: 60px 0 100px 0;
  }

  ${media.sm`
    ${SeriesAboutContainer} {
      margin: 40px 0 100px 0;
    }
  `};
`
