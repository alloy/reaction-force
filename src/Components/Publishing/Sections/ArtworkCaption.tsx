import _ from "lodash"
import React from "react"
import styled, { StyledFunction } from "styled-components"
import Colors from "../../../Assets/Colors"
import { pMedia } from "../../Helpers"
import TextLink from "../../TextLink"
import { Fonts } from "../Fonts"
import { Layout, SectionLayout } from "../Typings"
import { Truncator } from "./Truncator"

interface ArtworkCaptionProps extends React.HTMLProps<HTMLDivElement> {
  artwork: any
  layout?: Layout
  sectionLayout?: SectionLayout
  linked?: boolean
  isFullscreenCaption?: boolean
}

interface StyledArtworkCaptionProps {
  layout?: Layout
  sectionLayout?: SectionLayout
}

export class ArtworkCaption extends React.Component<ArtworkCaptionProps, null> {
  joinParts(children, delimiter = ", ") {
    const joined = _.compact(children).reduce((prev, curr) => {
      return [prev, delimiter, curr]
    }, [])
    return joined
  }

  renderArtists() {
    const { artwork: { artist, artists } } = this.props

    // Multiple artists
    if (artists && artists.length > 0) {
      const names = artists.map((a, i) => {
        const artistName = this.renderArtistName(a, i)
        return artistName
      })

      const joinedNames = this.joinParts(names)
      return joinedNames

      // Single artist
    } else if (artist) {
      const artistName = this.renderArtistName(artist, 0)
      return artistName
    }
  }

  renderArtistName(artist, i) {
    const { linked } = this.props
    const { name, slug } = artist
    const createTextLink = linked && slug

    if (createTextLink) {
      const href = `/artist/${slug}`

      return (
        <span key={`artist-${i}`} className="name">
          <TextLink href={href} color="#999">
            {name}
          </TextLink>
        </span>
      )
    } else {
      return <span className="name">{name}</span>
    }
  }

  renderTitleDate() {
    const children = [this.renderTitle(), this.renderDate()]

    const titleDate = this.joinParts(children)
    return titleDate
  }

  renderTitle() {
    const { artwork: { slug, title }, linked } = this.props

    if (title) {
      if (linked) {
        const href = `/artwork/${slug}`

        return (
          <span key={0} className="title">
            <TextLink href={href} color="#999">
              {title}
            </TextLink>
          </span>
        )
      } else {
        return (
          <span key={0} className="title">
            {title}
          </span>
        )
      }
    }
  }

  renderDate() {
    const { artwork: { date } } = this.props

    if (date && date.length) {
      return (
        <span key={1} className="date">
          {date}
        </span>
      )
    }
  }

  renderPartner() {
    const { artwork: { partner: { name, slug } }, linked } = this.props

    if (name) {
      const createTextLink = Boolean(linked && slug)

      if (createTextLink) {
        const href = `/partner/${slug}`

        return (
          <TextLink key={2} href={href} color="#999">
            {name}
          </TextLink>
        )
      } else {
        return name
      }
    }
  }

  renderCredit() {
    const { artwork: { credit } } = this.props

    if (credit && credit.length) {
      return (
        <span key={3} className="credit">
          {credit}
        </span>
      )
    }
  }

  renderPartnerCredit = () => {
    const children = [this.renderPartner(), this.renderCredit()]

    const joined = this.joinParts(children, ". ")
    return joined
  }

  render() {
    const { layout, isFullscreenCaption, sectionLayout } = this.props

    // Fullscreen
    if (isFullscreenCaption) {
      return (
        <StyledFullscreenCaption layout={layout}>
          <Line>
            <ArtistName>{this.renderArtists()}</ArtistName>
          </Line>
          <div>
            <Line>{this.renderTitleDate()}</Line>
            <Line>{this.renderPartnerCredit()}</Line>
          </div>
        </StyledFullscreenCaption>
      )

      // Classic Layout
    } else if (layout === "classic") {
      return (
        <StyledClassicCaption
          layout={layout}
          className="display-artwork__caption"
        >
          <Truncator>
            <ArtistName>{this.renderArtists()}</ArtistName>

            {this.renderTitleDate()}
            {". "}
            {this.renderPartner()}
          </Truncator>
        </StyledClassicCaption>
      )

      // Default (Standard + Feature)
    } else {
      return (
        <StyledArtworkCaption
          layout={layout}
          sectionLayout={sectionLayout}
          className="display-artwork__caption"
        >
          <ArtistName>{this.renderArtists()}</ArtistName>

          <div>
            <Truncator>{this.renderTitleDate()}</Truncator>

            <Truncator>{this.renderPartnerCredit()}</Truncator>
          </div>
        </StyledArtworkCaption>
      )
    }
  }
}

const ArtistName = styled.span`
  margin-right: 30px;
  white-space: nowrap;

  ${pMedia.xs`
    .artist-name {
      margin-right: 30px;
    }
  `};
`

const div: StyledFunction<
  StyledArtworkCaptionProps & React.HTMLProps<HTMLDivElement>
> =
  styled.div

const StyledArtworkCaption = div`
  padding: ${props => (props.sectionLayout === "fillwidth" ? "0 10px;" : "0;")}
  margin-top: 10px;
  display: flex;
  color: ${Colors.grayDark};

  ${Fonts.unica("s14")}
  .title {
    ${Fonts.unica("s14", "italic")}
  }

  ${pMedia.xs`
    padding: 0 10px;
  `}
`

const StyledClassicCaption = div`
  margin-top: 10px;
  display: block;
  color: ${Colors.grayDark};
  ${Fonts.garamond("s15")}

  ${ArtistName} {
    margin-right: 0;
    font-weight: bold;
    &:after {
      content: ', '
    }
  }

  .title {
    font-style: italic;
  }
`

const StyledFullscreenCaption = div`
  ${Fonts.unica("s16", "medium")}
  display: flex;
  color: black;

  .title {
    ${Fonts.unica("s16", "mediumItalic")}
  }

  ${pMedia.sm`
    ${Fonts.unica("s14", "medium")}
    flex-direction: column;

    .title {
      ${Fonts.unica("s14", "mediumItalic")}
    }
  `}
`

const Line = styled.div`
  ${pMedia.sm`
    &.artist-name {
      margin-bottom: 5px;
    }
  `};
`
