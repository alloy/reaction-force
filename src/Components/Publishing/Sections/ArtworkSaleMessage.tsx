import React from "react"
import * as Artsy from "Artsy"
import { Environment, QueryRenderer, graphql } from "react-relay"
import { ArtworkSaleMessageQuery } from "__generated__/ArtworkSaleMessageQuery.graphql"
import { get } from "Utils/get"

interface ArtworkSaleMessageProps {
  relayEnvironment: Environment
  artworkSlug: string
}

const ArtworkSaleMessage: React.FC<ArtworkSaleMessageProps> = props => {
  const saleMessageDisplay = artwork => {
    const { sale } = artwork
    const isAuction = sale && sale.isAuction

    if (isAuction) {
      const showBiddingClosed = sale.isClosed
      if (showBiddingClosed) {
        return "Bidding closed"
      } else {
        const highestBidDisplay = get(
          artwork,
          p => p.saleArtwork.highestBid.display
        )
        const openingBidDisplay = get(
          artwork,
          p => p.saleArtwork.openingBid.display
        )

        return highestBidDisplay || openingBidDisplay || ""
      }
    }

    return artwork.saleMessage === "Contact For Price"
      ? "Contact for price"
      : artwork.saleMessage
  }
  const { artworkSlug, relayEnvironment } = props
  return (
    <QueryRenderer<ArtworkSaleMessageQuery>
      environment={relayEnvironment}
      query={graphql`
        query ArtworkSaleMessageQuery($artworkSlug: String!) {
          artwork(id: $artworkSlug) {
            saleMessage
            sale {
              isAuction
              isClosed
            }
            saleArtwork {
              counts {
                bidderPositions
              }
              highestBid {
                display
              }
              openingBid {
                display
              }
            }
          }
        }
      `}
      variables={{ artworkSlug }}
      render={readyState => {
        const artwork = readyState?.props?.artwork
        return <div>{artwork ? saleMessageDisplay(artwork) : null}</div>
      }}
    />
  )
}

export const ArtworkSaleMessageContainer = Artsy.withSystemContext(
  ArtworkSaleMessage
)
