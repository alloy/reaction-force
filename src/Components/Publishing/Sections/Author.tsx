import React from "react"
import Markdown from "react-markdown"
import styled, { StyledFunction } from "styled-components"
import { resize } from "../../../Utils/resizer"
import { pMedia } from "../../Helpers"
import Icon from "../../Icon"
import { Fonts } from "../Fonts"

interface AuthorProps {
  author: any
}

export const Author: React.SFC<AuthorProps> = props => {
  const { author } = props
  const profileImage = author.image_url ? <ProfileImage src={resize(author.image_url, { width: 200 })} /> : false
  return (
    <AuthorContainer>
      {profileImage}
      <AuthorInfo>
        <Markdown source={author.bio} disallowedTypes={["Paragraph"]} unwrapDisallowed containerTagName="span" />
        <Twitter>
          <TwitterHandle href={`http://twitter.com/${author.twitter_handle}`}>
            <Icon name="twitter" color="black" />
            {`@${author.twitter_handle}`}
          </TwitterHandle>
        </Twitter>
      </AuthorInfo>
    </AuthorContainer>
  )
}

interface ProfileImageProps extends React.HTMLProps<HTMLDivElement> {
  image_url?: string
}

const Div: StyledFunction<ProfileImageProps> = styled.div

const ProfileImage = Div`
  min-width: 70px;
  min-height: 70px;
  border-radius: 50%;
  background: url(${props => props.src || ""}) no-repeat center center;
  background-size: cover;
  margin-right: 10px;
  ${pMedia.xs`
    min-width: 40px;
    min-height: 40px;
  `}
`
const AuthorContainer = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: 20px;
`
const AuthorInfo = styled.div`
  display: block;
  ${Fonts.garamond("s23")}
  a {
    color: black;
  }
  ${pMedia.xs`
    ${Fonts.garamond("s17")}
  `}
`
const Twitter = styled.span`
  margin-left: 20px;
`
const TwitterHandle = styled.a`
  ${Fonts.unica("s14", "medium")}
  color: black;
  text-decoration: none;
  white-space: nowrap;
  ${Icon} {
    vertical-align: middle;
    margin: 0px;
  }
  ${pMedia.xs`
    ${Fonts.unica("s12", "medium")}
  `}
`
