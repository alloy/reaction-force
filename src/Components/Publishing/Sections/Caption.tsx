import { garamond, unica } from "Assets/Fonts"
import React from "react"
import styled, { StyledFunction } from "styled-components"
import { pMedia } from "../../Helpers"
import { ArticleLayout, SectionLayout } from "../Typings"

interface CaptionProps {
  caption: string
  index?: any
  layout?: ArticleLayout
  sectionLayout?: SectionLayout
}

interface FigcaptionProps {
  layout: ArticleLayout
  sectionLayout?: SectionLayout
}

export const Caption: React.SFC<CaptionProps> = props => {
  const { caption, children, layout, sectionLayout } = props

  const child = children || (
    <div dangerouslySetInnerHTML={{ __html: caption }} />
  )

  return (
    <CaptionContainer>
      <Figcaption layout={layout} sectionLayout={sectionLayout}>
        {child}
      </Figcaption>
    </CaptionContainer>
  )
}

export const CaptionContainer = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 10px 0 10px 0;
  ${pMedia.xs`
    padding: 0px 10px;
  `};
`
const div: StyledFunction<FigcaptionProps & React.HTMLProps<HTMLDivElement>> =
  styled.div

// includes draft placeholder class for editable text in Writer
const Figcaption = div`
  padding: ${props => (props.sectionLayout === "fillwidth" ? "0 10px;" : "0;")}
  width: 100%;
  word-break: break-word;

  & > p, p,
  .public-DraftEditorPlaceholder-root,
  .public-DraftStyleDefault-block {
    ${props => (props.layout === "classic" ? garamond("s15") : unica("s14"))}
    color: ${props => (props.layout === "classic" ? "#666" : "#999")};
    margin: 0;
  }

  & > a, a {
    color: #999;
    &:hover {
      color: black;
    }
  }

  ${pMedia.xs`
    padding: 0px;
  `}
`
