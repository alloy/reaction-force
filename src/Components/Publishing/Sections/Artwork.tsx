import React from "react"
import styled from "styled-components"
import { resize } from "../../../Utils/resizer"
import { Layout, SectionLayout } from "../Typings"
import ArtworkCaption from "./ArtworkCaption"
import ImageWrapper from "./ImageWrapper"

interface ArtworkProps {
  artwork: any
  layout?: Layout
  sectionLayout?: SectionLayout
  linked?: boolean
  width?: string | number
  height?: string | number
  images?: [object]
  slideIndex?: number
}

const ArtworkImage: React.SFC<ArtworkProps> = props => {
  const { artwork, linked, height, width, layout } = props
  const src = resize(artwork.image, { width: 1200 })
  const image = (
    <ImageWrapper
      images={props.images}
      slideIndex={props.slideIndex}
      layout={layout}
      src={src}
      className="display-artwork__image"
      width={width}
      height={height}
      alt={artwork.title}
      index={artwork.index}
    />
  )
  if (linked) {
    return <ArtworkImageLink href={"/artwork/" + artwork.slug}>{image}</ArtworkImageLink>
  } else {
    return image
  }
}

ArtworkImage.defaultProps = {
  width: "100%",
  height: "auto",
}

const Artwork: React.SFC<ArtworkProps> = props => {
  return (
    <div className="display-artwork">
      <ArtworkImage {...props} />
      <ArtworkCaption {...props} />
    </div>
  )
}

Artwork.defaultProps = {
  linked: true,
}

const ArtworkImageLink = styled.a`
  text-decoration: none;
`

export default Artwork
