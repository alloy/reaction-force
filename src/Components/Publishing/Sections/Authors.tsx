import React from "react"
import styled from "styled-components"
import { Author } from "./Author"

interface AuthorsProps {
  authors: Array<object>
}

export const Authors: React.SFC<AuthorsProps> = props => {
  const { authors } = props
  return <AuthorsContainer>{authors.map((author, i) => <Author author={author} key={i} />)}</AuthorsContainer>
}
const AuthorsContainer = styled.div`
  display: flex;
  flex-direction: column;
`
