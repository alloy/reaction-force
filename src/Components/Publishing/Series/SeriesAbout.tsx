import { Box, Flex, Sans } from "@artsy/palette"
import React, { Component } from "react"
import track, { TrackingProp } from "react-tracking"
import styled from "styled-components"

import { PartnerBlock } from "Components/Publishing/Partner/PartnerBlock"
import { Text } from "Components/Publishing/Sections/Text"
import { ArticleData } from "Components/Publishing/Typings"
import { Media } from "Utils/Responsive"

interface Props {
  article?: ArticleData
  color?: string
  editDescription?: any
  editSubTitle?: any
  tracking?: TrackingProp
}

@track()
export class SeriesAbout extends Component<Props, null> {
  public static defaultProps: Partial<Props>

  componentDidMount() {
    const textLink = document.querySelector(".SeriesAbout__description a")
    if (textLink) {
      textLink.addEventListener("click", this.onClickFooterLink)
    }
  }

  onClickFooterLink = event => {
    this.props.tracking.trackEvent({
      action: "Click",
      flow: "Partner Footer CTA",
      type: "external_link",
      destination_path: event.currentTarget.href,
    })
  }

  partnerBlock = () => {
    const {
      article: { sponsor },
      color,
    } = this.props
    const sponsorLogo =
      sponsor &&
      (color === "white"
        ? sponsor.partner_light_logo
        : sponsor.partner_dark_logo)

    return (
      <PartnerBlock
        logo={sponsorLogo}
        url={sponsor.partner_logo_link}
        trackingData={{
          type: "external link",
          destination_path: sponsor.partner_logo_link,
        }}
      />
    )
  }

  render() {
    const {
      article: { series, sponsor },
      color,
      editDescription,
      editSubTitle,
    } = this.props

    return (
      <SeriesAboutContainer
        color={color || "black"}
        maxWidth="1200px"
        mx="auto"
        width="100%"
        flexDirection={["column", "column", "row", "row"]}
      >
        <Flex
          width={[1, 1, 1 / 3, 1 / 3]}
          justifyContent="space-between"
          flexDirection="column"
        >
          <Sans size="8" mb={["20px", "20px", 0, 0]}>
            {editSubTitle
              ? editSubTitle
              : (series && series.sub_title) || "About the Series"}
          </Sans>

          <Media greaterThanOrEqual="md">
            {sponsor && <Box mb={1}>{this.partnerBlock()}</Box>}
          </Media>
        </Flex>

        <Flex width={[1, 1, 2 / 3, 2 / 3]} flexDirection="column">
          {editDescription ? (
            <Text layout="standard" color={color || "black"}>
              {editDescription}
            </Text>
          ) : (
            <div className="SeriesAbout__description">
              <Text
                layout="standard"
                color={color || "black"}
                html={series && series.description}
              />
            </div>
          )}
          <Media lessThan="md">
            {sponsor && <Box mt={60}>{this.partnerBlock()}</Box>}
          </Media>
        </Flex>
      </SeriesAboutContainer>
    )
  }
}

export const SeriesAboutContainer = styled(Flex)``
