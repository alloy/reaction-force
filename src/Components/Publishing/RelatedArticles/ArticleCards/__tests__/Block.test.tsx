import { clone } from "lodash"
import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import {
  SeriesArticleSponsored,
  StandardArticle,
  VideoArticle,
  FeatureArticle,
} from "../../../Fixtures/Articles"
import { ArticleCard } from "../ArticleCard"
import { ArticleCardsBlock } from "../Block"
import { SeriesAbout } from "../../../Series/SeriesAbout"
import { VerticalOrSeriesTitle } from "../../../Sections/VerticalOrSeriesTitle"

jest.mock("../../../../../Utils/track.ts", () => ({
  track: () => jest.fn(c => c),
}))

describe("ArticleCard", () => {
  it("renders properly", () => {
    const component = renderer
      .create(
        <ArticleCardsBlock
          article={StandardArticle}
          relatedArticles={[StandardArticle, VideoArticle]}
        />
      )
      .toJSON()

    expect(component).toMatchSnapshot()
  })

  it("renders a list of articles properly", () => {
    const component = mount(
      <ArticleCardsBlock
        article={StandardArticle}
        relatedArticles={[StandardArticle, VideoArticle]}
      />
    )
    expect(component.text()).toMatch(StandardArticle.vertical.name)
    expect(component.find(ArticleCard).length).toBe(2)
  })

  it("renders a list of articles in a series properly", () => {
    const FeatureSeriesArticle = clone({
      ...FeatureArticle,
      seriesArticle: SeriesArticleSponsored,
    })
    const component = mount(
      <ArticleCardsBlock
        article={FeatureSeriesArticle}
        relatedArticles={[StandardArticle, VideoArticle]}
      />
    )
    expect(component.find(VerticalOrSeriesTitle).length).toBe(1)
    expect(component.find(VerticalOrSeriesTitle).text()).toMatch(
      SeriesArticleSponsored.title
    )
    expect(component.find(ArticleCard).length).toBe(2)
    expect(component.find(SeriesAbout).length).toBe(1)
  })
})
