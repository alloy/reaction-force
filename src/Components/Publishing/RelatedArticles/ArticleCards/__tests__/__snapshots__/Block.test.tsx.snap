// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard renders properly 1`] = `
.c7 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  color: black;
}

.c16 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
}

.c6 a {
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.c4 {
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-right: 0.5rem;
  padding-left: 0.5rem;
}

.c9 {
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-right: 0.5rem;
  padding-left: 0.5rem;
}

.c17 {
  margin: 10px 20px 0 0;
}

.c17::before {
  content: "";
  display: inline-block;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin: 0 5px 1px 0;
  background-color: black;
}

.c18 {
  margin: 5px 20px 0 0;
  white-space: nowrap;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  color: black;
}

.c25 {
  width: 32px;
  height: 32px;
}

.c22 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
  -webkit-transition: opacity 0.15s;
  transition: opacity 0.15s;
  display: block;
}

.c20 {
  position: relative;
  width: 50%;
  height: inherit;
  margin-left: 30px;
}

.c11 {
  border: 1px solid;
  border-radius: 2px;
  color: black;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 .c21 {
  opacity: 1;
}

.c11:hover .c21 {
  opacity: 0.7;
}

.c11 .c19 {
  background: white;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 50%;
  margin-bottom: 5px;
}

.c13 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 45px;
  line-height: 1.2em;
  margin-bottom: 30px;
}

.c14 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 23px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
}

.c24 {
  position: absolute;
  left: 20px;
  bottom: 20px;
  right: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c23 svg {
  width: 40px;
}

.c2 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 12px;
}

.c8 {
  color: black;
}

.c8 .c10 {
  margin-bottom: 60px;
}

.c0 {
  color: black;
}

.c0 .c1 {
  margin: auto;
}

.c0 .c5 {
  font-family: Unica77LLWebRegular,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 32px;
  line-height: 1.1em;
  width: 100%;
  margin-bottom: 40px;
}

.c0 .c5 a {
  border-bottom: 2px solid;
}

.c0 .SeriesAbout__SeriesAboutContainer-s1xh0wwc-0 {
  margin: 60px 0 100px 0;
}

@media only screen and (min-width:0em) {
  .c9 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
    display: block;
  }
}

@media (max-width:900px) {
  .c20 {
    width: 100%;
    margin-left: 0;
    margin-bottom: 10px;
  }
}

@media (max-width:900px) {
  .c11 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    padding: 20px;
  }
}

@media (max-width:900px) {
  .c12 {
    width: 100%;
    margin-bottom: 0;
  }
}

@media (max-width:900px) {
  .c13 {
    font-family: Unica77LLWebRegular,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 32px;
    line-height: 1.1em;
  }
}

@media (max-width:900px) {
  .c14 {
    font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
    font-size: 19px;
    line-height: 1.5em;
    -webkit-font-smoothing: antialiased;
    margin-bottom: 20px;
  }
}

@media (max-width:900px) {
  .c23 svg {
    width: 30px;
  }
}

@media (max-width:64em) {
  .c8 .c10 {
    margin-bottom: 40px;
  }
}

@media (max-width:48em) {
  .c0 .c5 a {
    display: block;
  }
}

@media (max-width:48em) {
  .c0 .SeriesAbout__SeriesAboutContainer-s1xh0wwc-0 {
    margin: 40px 0 100px 0;
  }
}

<div
  className="c0"
  color="black"
>
  <div
    className="c1 c2 c3"
  >
    <div
      className="c4"
    >
      <div
        className="c5 c6 c7"
        color="black"
        fontFamily={
          Object {
            "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
            "fontWeight": 500,
          }
        }
        fontSize="14px"
      >
        More in 
        <span>
          Art Market
        </span>
      </div>
    </div>
    <div
      className="c8"
      color="black"
    >
      <div
        className="c9"
      >
        <a
          className="ArticleCard c10 c11"
          color="black"
          href="new-yorks-next-art-district"
          onClick={[Function]}
        >
          <div
            className="c12"
          >
            <div>
              <div
                className="c13"
              >
                New York's Next Art District
              </div>
              <div
                className="c14"
              >
                Land exhibitions, make influential contacts, and gain valuable feedback about your work.
              </div>
            </div>
            <div
              className="Byline c15"
              color="black"
            >
              <div
                className="c16"
                fontFamily={
                  Object {
                    "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                    "fontWeight": 500,
                  }
                }
                fontSize="14px"
              >
                <div
                  className="c17"
                  color="black"
                >
                  Casey Lesser
                </div>
              </div>
              <div
                className="c18"
              >
                <div
                  className="c16"
                  fontFamily={
                    Object {
                      "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                      "fontWeight": 500,
                    }
                  }
                  fontSize="14px"
                >
                  May 19, 2017
                </div>
              </div>
            </div>
          </div>
          <div
            className="c19 c20"
          >
            <img
              className="c21 c22"
              src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2F7lsxxsw0qPAuKl37jEYitw%252Farticle%2Basset%2B1-hig%2Bres%2Bcopy.jpg&width=680&height=450&quality=95"
            />
          </div>
        </a>
      </div>
      <div
        className="c9"
      >
        <a
          className="ArticleCard c10 c11"
          color="black"
          href="joanne-artman-gallery-poetry-naturerefinement-form"
          onClick={[Function]}
        >
          <div
            className="c12"
          >
            <div>
              <div
                className="c13"
              >
                New Study Shows the Gender Pay Gap for Artists Is Not So Simple
              </div>
              <div
                className="c14"
              >
                The elegant spiral of the Nautilus shell, the sinuous pattern of the banks of a river, or the swirling vortex street of clouds - patterns exist.
              </div>
            </div>
            <div
              className="c18"
            >
              <div
                className="c16"
                fontFamily={
                  Object {
                    "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                    "fontWeight": 500,
                  }
                }
                fontSize="14px"
              >
                Aug 28, 2017
              </div>
            </div>
          </div>
          <div
            className="c19 c20"
          >
            <img
              className="c21 c22"
              src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2FwHFgQlrTrHav5O6bQRJ0dg%252FUntitled%2BSuspended_30x67x33%2B%25282%2529_sm%2Bcropped.jpg&width=680&height=450&quality=95"
            />
            <div
              className="c23 c24"
            >
              <svg
                className="c25"
                version="1.1"
                viewBox="0 0 40 56"
                x="0px"
                xmlns="http://www.w3.org/2000/svg"
                y="0px"
              >
                <g
                  fill="none"
                  fillRule="evenodd"
                  stroke="none"
                  strokeWidth="1"
                >
                  <polygon
                    fill="black"
                    points="0 0 0 56 39.3600006 28"
                  />
                </g>
              </svg>
              <div
                className="c16"
                fontFamily={
                  Object {
                    "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                    "fontWeight": 500,
                  }
                }
                fontSize="14px"
              >
                02:28
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
